src/realm/object-store/impl/transact_log_handler.cpp

```diff
@@ -130,7 +133,8 @@ void KVOAdapter::before(Transaction& sg)
             continue;
         }
         // otherwise the column should have been marked as modified
-        auto it = list.observer->changes.find(list.col.value);
+        REALM_ASSERT(list.path[0].is_col_key()); // first element in path is always the col_key
+        auto it = list.observer->changes.find(list.path[0].get_col_key().value);
```

Comment:
Same as above

Reply:
I think this works because changes are still indexed by col_key, but it has a nasty effect of notifying N times, where N is the number of collections that are associated to the col_key.